### DISCUSSION TITLE
How to suppress this error "expected an integer constant C/C++(661)" ?

### DISCUSSION THREAD

**Discussion by huangda1982** [microsoft/vscode-cpptools#12309] [Q&A] [answered]
Code is like this:
`#pragma pack`

This is legal with the compiler I am using. I can compile it with tasks.json.

Why can VSCode show compiling errors? Dose VSCode compile my c file with some internal compiler to check compiling errors?
How to tell VSCode not to check this error?
How to tell VSCode to check compiling errors with my compiler?

**Comment by sean-mcmanus** [comment]
What compiler? You could use the Select IntelliSense Configuration command to select a compiler.

C/C++(661) is an IntelliSense error.

Or use

```
#ifdef __INTELLISENSE__
    #pragma diag_suppress 661
#endif
```

if you only want to suppress that one error.

**Reply by huangda1982** [reply]
Thank you for your reply.

The compiler is called "CCRX", provided by Renesas for their RX series MCU.
It said 
`Unable to resolve configuration with compilerPath "c:\Program Files (x86)\Renesas\RX\3_3_0\bin\ccrx.exe".  Using "cl.exe" instead.`
when I selected the compiler.

Is there any way to avoid changing source code?

**Reply by sean-mcmanus** [reply, marked as answer]
@huangda1982 You could probably add that code to a file and then set the file in the forcedInclude property. Or you could set C_Cpp.intelliSenseEngine to "Tag Parser" to turn off all IntelliSense features.
